nbconvert/exporters/script.py

```diff
@@ -19,15 +18,29 @@ def _template_file_default(self):
 
     def from_notebook_node(self, nb, resources=None, **kw):
         langinfo = nb.metadata.get('language_info', {})
-        
         # delegate to custom exporter, if specified
         exporter_name = langinfo.get('nbconvert_exporter')
         if exporter_name and exporter_name != 'script':
             self.log.debug("Loading script exporter: %s", exporter_name)
-            from .export import get_exporter
+            from .export import get_exporter, ExporterNameError
             if exporter_name not in self._exporters:
-                Exporter = get_exporter(exporter_name)
-                self._exporters[exporter_name] = Exporter(parent=self)
+                try: 
+                    Exporter = get_exporter(exporter_name)
+                    self._exporters[exporter_name] = Exporter(parent=self)
+                except ValueError:
+                    if langinfo.get('nbconvert_exporter_package',""):
+                        external_exporter_name = langinfo.get('nbconvert_exporter_package')
+                        try: 
+                            Exporter = get_exporter(external_exporter_name)
```

Comment:
I'm not clear what you're suggesting as the alternative.

One other way we could do it is for `ScriptExporter` to look in a second group of entry points, like `nbconvert.exporters.script`, keyed by the language name already saved in the notebook. That would probably be more like my other suggestion for this issue: there's no suggestion to install a package, but it falls back to `ScriptExporter` if it can't find a more specific exporter.

 We should keep supporting the existing metadata field for some time even if we do deprecate it, because it's explicitly public API.